tests/fit_data/test_fit_data_set_cells.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+
+from eddington import FitData
+from eddington.exceptions import FitDataColumnsSelectionError, FitDataInvalidSyntax, FitDataColumnAlreadyExists
+
+from tests.fit_data import COLUMNS, NUMBER_OF_RECORDS, VALUES
+
+import numpy as np
+import string
+
+
+def set_cell_allowed():
+    fit_data = FitData(COLUMNS)
+    rows = np.random.randint(1, NUMBER_OF_RECORDS, size=(4, 7))
+    values = np.random.rand(4, 7)
+
+    for i in range(4):
+        for j in range(7):
+            fit_data.set_cell(rows[i][j], COLUMNS.keys()[i], values[i][j])
+
+    assert fit_data.x[rows] == values[0], f'Some of the cells did not change as expected:\nactual values:"{fit_data.x[rows]}"\nexpected values:"{values[0]}"'
+    assert fit_data.x_err[rows] == values[1], f'Some of the cells did not change as expected:\nactual values:"{fit_data.x_err[rows]}"\nexpected values:"{values[1]}"'
+    assert fit_data.y[rows] == values[2], f'Some of the cells did not change as expected:\nactual values:"{fit_data.y[rows]}"\nexpected values:"{values[2]}"'
+    assert fit_data.y_err[rows] == values[3], f'Some of the cells did not change as expected:\nactual values:"{fit_data.y_err[rows]}"\nexpected values:"{values[3]}"'
+
+def set_cell_not_allowed():
+    fit_data = FitData(COLUMNS)
+    rows = np.random.randint(1, NUMBER_OF_RECORDS, size=(4, 2))
+    values = [[''.join(np.random.choice(string.ascii_letters) for i in range(3)),
+        ''.join(np.random.choice(string.punctuation) for i in range(3))] for i in range(4)]
+
+    for i in range(4):
+        for j in range(2):
+            with pytest.raises(FitDataInvalidSyntax):
+                fit_data.set_cell(rows[i][j], COLUMNS.keys()[i], values[i][j])
+
+def set_header_allowed():
+    fit_data = FitData(COLUMNS)
+
+    new_x = ''.join(np.random.choice(string.ascii_letters) for i in range(3))
```

Comment:
If you use a piece of code more than 3 times, it's worth making it a method, even in test files.
Here, you use `''.join(np.random.choice(string.ascii_letters) for i in range(n))` quite alot. you should add the following method:
```python
def random_name(n):
    return ''.join(np.random.choice(string.ascii_letters) for i in range(n))
```
 and reuse it when possible :)